test/demo/eventListeners.inttest.ts

```diff
@@ -0,0 +1,170 @@
+                expect(buttonHandles).toHaveLength(1);
+
+                // Click on the button and pause momentarily to allow the event to propagate
+                await buttonHandles[0].click();
+                await delay(100);
+
+                // buttonClickListener triggered with correct metadata
+                expect(buttonClickListener).toHaveBeenCalledTimes(1);
+                expect(buttonClickListener).toHaveBeenCalledWith({
+                    type: 'notification-button-clicked',
+                    notification: note,
+                    buttonIndex: 0
+                });
+
+                // clickListener not triggered
+                expect(clickListener).toHaveBeenCalledTimes(0);
+            });
+
+            describe('When clicking the close button', () => {
+                beforeEach(async () => {
+                    const noteCards = await getCardsByNotification(testApp.identity.uuid, note.id);
+
+                    // Close button is only visible/clickable when card is hovered
+                    await noteCards[0].hover();
+
+                    // Get a remote handle to the close button DOM element
+                    const closeHandles = await noteCards[0].$$('.close');
+                    expect(closeHandles).toHaveLength(1);
+
+                    // Click on the button and pause momentarily to allow the event to propagate
+                    await closeHandles[0].click();
+                    await delay(100);
+                });
+
+                test('The closeListener is called once with the correct metadata the other listeners are not called', async () => {
+                    expect(closeListener).toHaveBeenCalledTimes(1);
+                    expect(closeListener).toHaveBeenCalledWith({
+                        type: 'notification-closed',
+                        notification: note
+                    });
+
+                    // Other listeners not triggered
+                    expect(clickListener).toHaveBeenCalledTimes(0);
+                    expect(buttonClickListener).toHaveBeenCalledTimes(0);
+                });
+
+                test('The notification is cleared and no longer appears in the center or when calling getAll', async () => {
+                    // No card in the center
+                    const noteCards = await getCardsByNotification(testApp.identity.uuid, note.id);
+                    expect(noteCards).toHaveLength(0);
+
+                    // Not returned from getAll
+                    const appNotes = await notifsRemote.getAll(testAppMainWindow.identity);
+                    expect(appNotes).not.toContainEqual(note);
+                });
+            });
+        });
+    });
+});
+
+
+// TODO: Make this a util modeled on layouts' spawn utils (SERVICE-524)
+const nextUuid = (() => {
+    let count = 100;
+    return () => 'notifications-test-app-' + (count++);
+})();
+
+async function createTestApp():Promise<Application> {
```

Comment:
Even if this util is temporary, can we move it to a separate file rather than duplicating?

Reply:
done